libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/roomlist/RoomSummaryListProcessor.kt

```diff
@@ -134,7 +134,19 @@ class RoomSummaryListProcessor(
         mutex.withLock {
             val mutableRoomSummaries = roomSummaries.value.toMutableList()
             block(mutableRoomSummaries)
-            roomSummaries.value = mutableRoomSummaries
+            // Ensure that there is no duplicated rooms.
+            val duplicatedRoomIds = mutableRoomSummaries
+                .map { it.identifier() }
+                .groupingBy { it }
+                .eachCount()
+                .filter { it.value > 1 }
+                .keys
+            if (duplicatedRoomIds.isNotEmpty()) {
```

Comment:
I'd just de-deduplicate right away to avoid looping twice over the all list...
We could also let the ui just filter it later as we already loop through all room list items later

Reply:
Also, this is not related to this PR can we do that elsewhere?

Reply:
But this shouldn't crash the UI as we don't use the key parameter on this lazy column :

 ```
itemsIndexed(
                    items = state.roomList,
                    contentType = { _, room -> room.contentType() },
                ) { index, room ->
                    RoomSummaryRow(
                        room = room,
                        onClick = ::onRoomClicked,
                        onLongClick = onRoomLongClicked,
                    )
                    if (index != state.roomList.lastIndex) {
                        HorizontalDivider()
                    }
                }
```

Reply:
Actually, this is a recent change, that explain why I did not observe the crash on a too old version: https://github.com/element-hq/element-x-android/commit/02fe07a6bdf631a1a935eba08af491a855dddeab#diff-6e3590fab2a1b58a5f64cd00a3c09123a6369e317092e489e601867d4496f6fbR265

Reply:
> Also, this is not related to this PR can we do that elsewhere?

I thought it was due to the SDK upgrade but actually it's not. Do you really want me to create a dedicated PR for this?

Reply:
> Actually, this is a recent change, that explain why I did not observe the crash on a too old version: [02fe07a#diff-6e3590fab2a1b58a5f64cd00a3c09123a6369e317092e489e601867d4496f6fbR265](https://github.com/element-hq/element-x-android/commit/02fe07a6bdf631a1a935eba08af491a855dddeab#diff-6e3590fab2a1b58a5f64cd00a3c09123a6369e317092e489e601867d4496f6fbR265)

But we shouldn't use `key` on the room list, otherwise it'll "break" the scroll position. 

So I'd create a new PR with both changes : 
- Revert using the key param
- De-deduplicate room in the sdk (hopefully performances are still ok with big room list? Otherwise we might just want to wait for the proper fix with room ordering rust sdk pr, as this is not new).

Reply:
OK, let's do this. Thanks